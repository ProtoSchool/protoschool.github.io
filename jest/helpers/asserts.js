// ** SNAPSHOT FILTERING ***
// prevent snapshots for logs that change on each instance, such as shuffled quiz options

const excludedPhrases = [
  "Cool! Here's what we get when we mix up the order of the answer choices:"
]

// returns true if log includes a string from the excludedPhrases array
function isExcluded (log) {
  return excludedPhrases.some(phrase => log.includes(phrase))
}

// add log to snapshot UNLESS it contains an excluded phrase
export function assertLogSnapshot (log) {
  if (!isExcluded(log)) {
    expect(log).toMatchSnapshot() // expect if log isn't excluded from assertion
  }
  // do nothing if log is excluded from assertion
}
