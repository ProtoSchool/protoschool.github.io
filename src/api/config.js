/* eslint-disable no-unused-expressions */
export const staticPath = process.env.DATA_STATIC_PATH || 'src/static'
export const tutorialsPath = process.env.DATA_TUTORIALS_PATH || 'src/tutorials'

const boilerplatesPath = process.env.LESSON_BOILERPLATES_PATH || `${tutorialsPath}/boilerplates`

export const boilerplates = {
  path: boilerplatesPath,
  markdownPath: `${boilerplatesPath}/boilerplate.md`,
  challengeMarkdownPath: `${boilerplatesPath}/boilerplate-challenge.md`,
  multipleChoiceJsPath: `${boilerplatesPath}/boilerplate-multiple-choice.js`,
  codeJsPath: `${boilerplatesPath}/boilerplate-code.js`,
  fileUploadJsPath: `${boilerplatesPath}/boilerplate-file-upload.js`
}
