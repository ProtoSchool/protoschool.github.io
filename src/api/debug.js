export default (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test') && process.env.DEBUG
