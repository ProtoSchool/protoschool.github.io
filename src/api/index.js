/**
 * API for ProtoSchool.
 *
 * TODO: Make API universal https://github.com/ProtoSchool/protoschool.github.io/issues/589
 *  - make all scripts use this API
 *  - change it to be universal so the website can also use it
 *   - ATM we have deps on fs.readFile etc
 *
 * @module api
 */
module.exports = {
  projects: require('./modules/projects'),
  lessons: require('./modules/lessons'),
  courses: require('./modules/courses').default,
  resources: require('./modules/resources'),
  tutorials: require('./modules/tutorials'),
  _config: require('./config')
}
