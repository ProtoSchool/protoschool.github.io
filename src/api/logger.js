export const debug = (...args) => {
  console.log('⚙️ api', ...args)
}

export const createLogGroup = module => functionMethod => `[${module}.${functionMethod}()]`
