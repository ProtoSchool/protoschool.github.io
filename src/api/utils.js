import { writeFileSync } from 'fs'
import { resolve } from 'path'
import { staticPath } from './config.js'

const correctedCases = {
  api: 'API',
  cid: 'CID',
  of: 'of',
  a: 'a'
}

export function writeStaticFile(staticFileName, data) {
  writeFileSync(resolve(staticPath, staticFileName), JSON.stringify(data, null, 2))
}

export function deriveShortname(path) {
  return path.split('-').map(word => (
    correctedCases[word] ? correctedCases[word] : (word.charAt(0).toUpperCase() + word.slice(1))
  )).join(' ')
}
