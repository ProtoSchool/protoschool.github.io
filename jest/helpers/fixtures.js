import { lessons as _lessons, resources as _resources, courses, tutorials } from '../../src/api'

// Creators: create data (tutorials, lessons)
export function createTutorial (config = { override: {}, lessons: 0, resources: 0 }) {
  const { tutorial, lessons, resources } = generateTutorial(config)
  const createdTutorial = tutorials.create(tutorial)

  for (let i = 0; i < lessons.length; ++i) {
    _lessons.create(tutorials.get(createdTutorial.id), lessons[i])
  }

  resources.forEach(resource => {
    _resources.add(createdTutorial.id, resource)
  })

  return tutorials.get(createdTutorial.id)
}

// Generators: generate data to be used

export function generateTutorial (config = { override: {}, lessons: 0, resources: 0, lessonOverride: {} }) {
  const suffix = tutorials.getNextTutorialId()

  const tutorial = {
    title: `New Tutorial (${suffix})`,
    url: `new-tut-${suffix}`,
    project: 'libp2p',
    description: 'New tutorial description',
    ...config.override
  }

  const lessons = new Array(config.lessons).fill().map((_, i) => (
    generateLesson({
      override: {
        title: `Lesson ${i + 1}`,
        ...config.lessonOverride
      }
    }).lesson
  ))

  const resources = new Array(config.resources).fill().map((_, i) => (
    generateResource({
      override: {
        title: `Resource ${i + 1}`,
        link: `https://resource${i + 1}.com`
      }
    }).resource
  ))

  return {
    tutorial,
    lessons,
    resources,
    expected: { ...tutorial, lessons, resources }
  }
}

export function generateLesson ({ createTutorial = false, override = {} } = {}) {
  const lesson = {
    title: 'Lesson',
    type: 'text',
    ...override
  }
  let tutorial

  if (createTutorial) {
    tutorial = tutorials.create(generateTutorial().tutorial)
    courses.add(tutorial.id)
  }

  return {
    lesson,
    tutorial,
    expected: { lesson, tutorial }
  }
}

export function generateResource ({ createTutorial, override = {} } = {}) {
  const resource = {
    title: 'Resource',
    link: 'https://resource.com',
    description: 'Resoure description',
    type: 'demo',
    ...override
  }
  let tutorial

  if (createTutorial) {
    tutorial = tutorials.create(generateTutorial().tutorial)
    courses.add(tutorial.id)
  }

  return {
    resource,
    tutorial,
    expected: { resource, tutorial }
  }
}
