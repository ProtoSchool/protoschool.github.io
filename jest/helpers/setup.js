import { courses, tutorials } from '../../src/api'

export function restoreData (lastTutorialId) {
  const newLastTutorialId = tutorials.list.getLatest().id

  // delete all new tutorials
  for (let id = lastTutorialId + 1; id <= newLastTutorialId; id++) {
    tutorials.remove(id)
    courses.remove(id)
  }
}
