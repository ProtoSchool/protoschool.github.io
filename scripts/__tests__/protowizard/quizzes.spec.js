import { tutorials, lessons as _lessons } from '../../../src/api'
import { restoreData } from '../../../jest/helpers/setup'
import { createTutorial } from '../../../jest/helpers/fixtures'
import { assertQuizUnknownOrder, assertQuizKnownOrder } from '../helpers/asserts'
import { protowizard } from '../helpers/runners'

const hardcodedData = {
  question: "What's my question?",
  choices: [
    {
      answer: 'Incorrect answer 1',
      correct: false,
      feedback: 'Feedback for incorrect answer 1'
    },
    {
      answer: 'Correct answer',
      correct: true,
      feedback: 'Positive feedback'
    },
    {
      answer: 'Incorrect answer 2',
      correct: false,
      feedback: 'Feedback for incorrect answer 2'
    }
  ]
}

describe('protowizard', () => {
  let lastTutorialId

  beforeAll(() => {
    lastTutorialId = tutorials.list.getLatest().id
  })

  afterEach(() => {
    restoreData(lastTutorialId)
  })

  describe('4. create quiz', () => {
    test('4.1 should add quiz to pristine lesson', async () => {
      const tutorial = createTutorial({
        lessons: 1,
        lessonOverride: {
          type: 'multiple-choice'
        }
      })

      await protowizard([
        { type: 'quiz' },
        { confirm: true }, // lesson already exists
        { latestTutorial: true }, // add to latest (test) tutorial
        { lesson: tutorial.lessons[0] }, // select first (test) lesson
        { confirm: true }, // overwrite pristine quiz
        { question: "What's my question?", // question
          correctAnswer: 'Correct answer', // correct answer
          correctFeedback: 'Positive feedback' }, // correct feedback
        { incorrectAnswer: 'Incorrect answer 1', // incorrect answer
          incorrectFeedback: 'Feedback for incorrect answer 1' }, // incorrect feedback
        { confirm: true }, // add another incorrect answer
        { incorrectAnswer: 'Incorrect answer 2', // incorrect answer
          incorrectFeedback: 'Feedback for incorrect answer 2' }, // incorrect feedback
        { confirm: false }, // don't add another answer
        { confirm: false } // don't add another quiz
      ])

      const result = tutorials.getByUrl(tutorial.url)

      expect(result.lessons).toHaveLength(1)

      assertQuizUnknownOrder({ result: result.lessons[0], hardcodedData })
    })

    test('4.2 should add quiz to existing lesson', async () => {
      const tutorial = createTutorial({
        lessons: 1,
        lessonOverride: {
          type: 'multiple-choice'
        }
      })

      await protowizard([
        // add quiz just to overwrite later
        { type: 'quiz' },
        { confirm: true }, // lesson already exists
        { latestTutorial: true }, // add to latest (test) tutorial
        { lesson: tutorial.lessons[0] }, // select first (test) lesson
        { confirm: true }, // overwrite pristine quiz
        { question: 'Question to overwrite', // question
          correctAnswer: 'Correct answer to overwrite', // correct answer
          correctFeedback: 'Correct feedback to overwrite' }, // correct feedback
        { incorrectAnswer: 'Incorrect answer to overwrite', // incorrect answer
          incorrectFeedback: 'Incorrect feedback to overwrite' }, // incorrect feedback
        { confirm: false }, // don't add another answer
        { confirm: true }, // add another quiz
        // add quiz and overwrite the one above
        { confirm: true }, // lesson already exists
        { latestTutorial: true }, // add to latest (test) tutorial
        { lesson: tutorial.lessons[0] }, // select first (test) lesson
        { confirm: true }, // overwrite non-pristine quiz
        { question: "What's my question?", // question
          correctAnswer: 'Correct answer', // correct answer
          correctFeedback: 'Positive feedback' }, // correct feedback
        { incorrectAnswer: 'Incorrect answer 1', // incorrect answer
          incorrectFeedback: 'Feedback for incorrect answer 1' }, // incorrect feedback
        { confirm: true }, // add another incorrect answer
        { incorrectAnswer: 'Incorrect answer 2', // incorrect answer
          incorrectFeedback: 'Feedback for incorrect answer 2' }, // incorrect feedback
        { confirm: false }, // don't add another answer
        { confirm: false } // don't add another quiz
      ])

      const result = tutorials.getByUrl(tutorial.url)

      expect(result.lessons).toHaveLength(1)
      assertQuizUnknownOrder({ result: result.lessons[0], hardcodedData })
    })

    test('4.3 should cancel overwrite of non-pristine lesson', async () => {
      const tutorial = createTutorial({
        lessons: 1,
        lessonOverride: {
          type: 'multiple-choice'
        }
      })

      // use API to create a non-pristine quiz
      _lessons.updateQuiz(tutorial, tutorial.lessons[0], hardcodedData)

      await protowizard([
        // start to make a new quiz and then cancel rather than overwrite non-pristine
        { type: 'quiz' },
        { confirm: true }, // lesson already exists
        { latestTutorial: true }, // add to latest (test) tutorial
        { lesson: tutorial.lessons[0] }, // select first (test) lesson
        { confirm: false } // cancel overwriting non-pristine quiz
      ])

      const result = tutorials.getByUrl(tutorial.url)

      expect(result.lessons).toHaveLength(1)

      assertQuizKnownOrder({ result })
    })

    test('4.4 should cancel overwrite of pristine lesson', async () => {
      const tutorial = createTutorial({
        lessons: 1,
        lessonOverride: {
          type: 'multiple-choice'
        }
      })

      await protowizard([
        { type: 'quiz' },
        { confirm: true }, // lesson already exists
        { latestTutorial: true }, // add to latest (test) tutorial
        { lesson: tutorial.lessons[0] }, // select first (test) lesson
        { confirm: false } // cancel overwriting pristine quiz
      ])
    })
  })
})
