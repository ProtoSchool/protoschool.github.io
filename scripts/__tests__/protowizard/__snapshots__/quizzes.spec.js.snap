// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 1`] = `"Let's create a multiple-choice quiz! This will only work if you've already created the tutorial and lesson files."`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 2`] = `
"Great! You've chosen to build a quiz for:
 ‣ Tutorial: New Tutorial (11)
 ‣ Lesson: Lesson 1"
`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 3`] = `"When we create a quiz together, it will overwrite any existing content. If you think you may have previously updated this quiz manually, you may want to double check before proceeding."`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 4`] = `"To preview the quiz in its current state, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 5`] = `"Now it's time to write the question for your multiple-choice quiz and provide answer choices, with positive or negative feedback for each. Wrong answers, and the feedback associated with them, are a great way to address common misconceptions about the topic. Be sure to make your feedback as helpful as possible, to guide the learner to the right choice. You'll need to create 1 correct answer and 2-3 incorrect answers. (I'll take care of randomizing their order later.)"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 6`] = `"Question: \\"What's my question?\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 7`] = `"Answer (Correct): \\"Correct answer\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 8`] = `"Feedback: \\"Positive feedback\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 9`] = `"Answer (Incorrect): \\"Incorrect answer 1\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 10`] = `"Feedback: \\"Feedback for incorrect answer 1\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 11`] = `"You currently have 1 correct answer and 1 wrong answer. (I recommend providing 2-3 incorrect options.)"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 12`] = `"Answer (Incorrect): \\"Incorrect answer 2\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 13`] = `"Feedback: \\"Feedback for incorrect answer 2\\""`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 14`] = `"You currently have 1 correct answer and 2 wrong answers. (I recommend providing 2-3 incorrect options.)"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 15`] = `"To preview your quiz, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 16`] = `"To make changes to your quiz, you can edit this file directly: src/tutorials/0011-new-tut-11/01.js"`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 17`] = `"Okay, no problem. You can summon me later to add  more quizzes."`;

exports[`protowizard 4. create quiz 4.1 should add quiz to pristine lesson 18`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 2`] = `"Let's create a multiple-choice quiz! This will only work if you've already created the tutorial and lesson files."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 3`] = `
"Great! You've chosen to build a quiz for:
 ‣ Tutorial: New Tutorial (11)
 ‣ Lesson: Lesson 1"
`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 4`] = `"When we create a quiz together, it will overwrite any existing content. If you think you may have previously updated this quiz manually, you may want to double check before proceeding."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 5`] = `"To preview the quiz in its current state, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 6`] = `"Now it's time to write the question for your multiple-choice quiz and provide answer choices, with positive or negative feedback for each. Wrong answers, and the feedback associated with them, are a great way to address common misconceptions about the topic. Be sure to make your feedback as helpful as possible, to guide the learner to the right choice. You'll need to create 1 correct answer and 2-3 incorrect answers. (I'll take care of randomizing their order later.)"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 7`] = `"Question: \\"Question to overwrite\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 8`] = `"Answer (Correct): \\"Correct answer to overwrite\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 9`] = `"Feedback: \\"Correct feedback to overwrite\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 10`] = `"Answer (Incorrect): \\"Incorrect answer to overwrite\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 11`] = `"Feedback: \\"Incorrect feedback to overwrite\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 12`] = `"You currently have 1 correct answer and 1 wrong answer. (I recommend providing 2-3 incorrect options.)"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 13`] = `"To preview your quiz, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 14`] = `"To make changes to your quiz, you can edit this file directly: src/tutorials/0011-new-tut-11/01.js"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 15`] = `"Let's create a multiple-choice quiz! This will only work if you've already created the tutorial and lesson files."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 16`] = `
"Great! You've chosen to build a quiz for:
 ‣ Tutorial: New Tutorial (11)
 ‣ Lesson: Lesson 1"
`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 17`] = `"It looks like you've already created the quiz for this lesson."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 18`] = `"To preview the existing version, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 19`] = `"Now it's time to write the question for your multiple-choice quiz and provide answer choices, with positive or negative feedback for each. Wrong answers, and the feedback associated with them, are a great way to address common misconceptions about the topic. Be sure to make your feedback as helpful as possible, to guide the learner to the right choice. You'll need to create 1 correct answer and 2-3 incorrect answers. (I'll take care of randomizing their order later.)"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 20`] = `"Question: \\"What's my question?\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 21`] = `"Answer (Correct): \\"Correct answer\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 22`] = `"Feedback: \\"Positive feedback\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 23`] = `"Answer (Incorrect): \\"Incorrect answer 1\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 24`] = `"Feedback: \\"Feedback for incorrect answer 1\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 25`] = `"You currently have 1 correct answer and 1 wrong answer. (I recommend providing 2-3 incorrect options.)"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 26`] = `"Answer (Incorrect): \\"Incorrect answer 2\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 27`] = `"Feedback: \\"Feedback for incorrect answer 2\\""`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 28`] = `"You currently have 1 correct answer and 2 wrong answers. (I recommend providing 2-3 incorrect options.)"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 29`] = `"To preview your quiz, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 30`] = `"To make changes to your quiz, you can edit this file directly: src/tutorials/0011-new-tut-11/01.js"`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 31`] = `"Okay, no problem. You can summon me later to add  more quizzes."`;

exports[`protowizard 4. create quiz 4.2 should add quiz to existing lesson 32`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 2`] = `"Let's create a multiple-choice quiz! This will only work if you've already created the tutorial and lesson files."`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 3`] = `
"Great! You've chosen to build a quiz for:
 ‣ Tutorial: New Tutorial (11)
 ‣ Lesson: Lesson 1"
`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 4`] = `"It looks like you've already created the quiz for this lesson."`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 5`] = `"To preview the existing version, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 6`] = `"Okay, no problem. You can edit your existing quiz directly in this file: src/tutorials/0011-new-tut-11/01.js"`;

exports[`protowizard 4. create quiz 4.3 should cancel overwrite of non-pristine lesson 7`] = `
"
/* eslint quotes: [\\"error\\", \\"double\\"]  */

// Question must be a string
const question = \\"What's my question?\\"

// Choices must be an array of objects, each with the properties:
// \`answer\` (string), \`correct\` (boolean), and \`feedback\` (string)
// Only one answer can be correct.
const choices = [
  {
    \\"answer\\": \\"Incorrect answer 1\\",
    \\"correct\\": false,
    \\"feedback\\": \\"Feedback for incorrect answer 1\\"
  },
  {
    \\"answer\\": \\"Correct answer\\",
    \\"correct\\": true,
    \\"feedback\\": \\"Positive feedback\\"
  },
  {
    \\"answer\\": \\"Incorrect answer 2\\",
    \\"correct\\": false,
    \\"feedback\\": \\"Feedback for incorrect answer 2\\"
  }
]

export default {
  question,
  choices
}
"
`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 2`] = `"Let's create a multiple-choice quiz! This will only work if you've already created the tutorial and lesson files."`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 3`] = `
"Great! You've chosen to build a quiz for:
 ‣ Tutorial: New Tutorial (11)
 ‣ Lesson: Lesson 1"
`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 4`] = `"When we create a quiz together, it will overwrite any existing content. If you think you may have previously updated this quiz manually, you may want to double check before proceeding."`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 5`] = `"To preview the quiz in its current state, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 4. create quiz 4.4 should cancel overwrite of pristine lesson 6`] = `"Okay, no problem. You can edit your existing quiz directly in this file: src/tutorials/0011-new-tut-11/01.js"`;
